src/main/java/li/cil/oc/util/ExtendedUnicodeHelper.java

```diff
@@ -0,0 +1,41 @@
+package li.cil.oc.util;
+
+/**
+ * Helper functions for handling strings with characters outside of the Unicode BMP.
+ */
+public final class ExtendedUnicodeHelper {
+	private ExtendedUnicodeHelper() {
+
+	}
+
+	public static int length(String s) {
+		return s.codePointCount(0, s.length());
+	}
+
+	public static String reverse(String s) {
```

Comment:
is this reverse code only used in string.reverse? have we tested s.reverse.reverse on invalid bytes? should s == s.reverse.reverse always?

Reply:
1. Yes, but it occurs twice (luac, luaj), and I didn't want to make the UnicodeAPI code even less understandable by throwing that implementation in there.
2. No, but I do wonder if they can even appear. Lua uses UTF-8, which is then converted to UTF-16 for Java Strings.
3. I'm not sure there's a standard, really? With this code, it should.